lib/crawler.js

```diff
@@ -477,18 +480,34 @@ class Crawler {
 
   // ===============  Helpers  ============
 
-  _queueDeadAndLog(request, reason) {
-    const error = new Error(reason);
-    error._type = request.type;
-    error._url = request.url;
-    error._cid = request.cid;
-    this.logger.error(error, request.meta);
-    return request.markDead('Archived', reason);
+  _storeDead(request, reason, error = null) {
+    request._error = error;
+    return request.markDead('Deadletter', reason);
+  }
+
+  storeDead(request, reason = null) {
+    const document = this._createDeadletter(request, reason);
+    return this.deadletters.upsert(document).then(() => {
+      debug(`_storeDeadletter(${request.meta.loopName}:${request.toUniqueString()}): exit (success)`);
```

Comment:
Prefix doesn't match the function name and we should have a debug on entry to the function.